### PR TITLE
Always run untag when removing docker image

### PR DESCRIPTION
**Pull request by corvus-ch**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

Under certain conditions a docker image can have one tag while
assiciated to multiple repositories. This prevents the image from being
removed due to the multi repository association. Always removing all
tags, prevents this from happening.

This can be reproduced with the following:

```bash
docker pull centos:7
docker tag centos:7 example.com:1234/centos:latest
docker push example.com:1234/centos:latest
docker rmi example.com:1234/centos:latest
docker pull example.com:1234/centos:latest
```

This ends with an image which has one image tag but multiple repo digests.

```bash
docker image inspect centos:7 | jq '.[0] | .RepoTags, .RepoDigests'
[
  "centos:latest"
]
[
  "centos@sha256:67dad89757a55bfdfabec8abd0e22f8c7c12a1856514726470228063ed86593b",
  "example.com:1234/centos@sha256:67dad89757a55bfdfabec8abd0e22f8c7c12a1856514726470228063ed86593b"
]
```

**Which issue(s) this PR fixes**
none

**Special notes for your reviewer**:
none

**Does this PR introduce a user-facing change?**:

```release-note
Image garbage collection no longer fails for images with only one tag but more than one repository associated.
```

### CODE REVIEW

File: pkg/kubelet/dockershim/docker_image.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (ds *dockerService) RemoveImage(_ context.Context, r *runtimeapi.RemoveImag
 	// TODO: We assume image.Image is image ID here, which is true in the current implementation
 	// of kubelet, but we should still clarify this in CRI.
 	imageInspect, err := ds.client.InspectImageByID(image.Image)
-	if err == nil && imageInspect != nil && len(imageInspect.RepoTags) > 1 {
+	if err == nil {
```
Comment (changyaowei) [inline pkg/kubelet/dockershim/docker_image.go]: Before that , kubelet remove it according to imageID , now we remove it to according to imageTag, does it work for you special condition ? and why ?

**Comment by corvus-ch** [comment]
Is there some action required on my side for this to proceed? If this is the case, please let my known and I will take care of it.

**Comment by mhutter** [comment]
@sjenning is there something more required before this can be merged and backported?

**Comment by sjenning** [comment]
needs lgtm and approval.  I can only lgtm.  This makes sense to me.
/lgtm

**Comment by sjenning** [comment]
/assign @derekwaynecarr @yujuhong 
for review/approval

**Comment by derekwaynecarr** [comment]
this makes sense to me as well.

/approve

File: pkg/kubelet/dockershim/docker_image.go [hunk 2]
```diff
@@ -120,24 +120,27 @@ func (ds *dockerService) RemoveImage(_ context.Context, r *runtimeapi.RemoveImag
...
+	// Iterating over both of them plus the image ID ensures the image really got removed.
+	// It also prevents images from being deleted, which actually are deletable using this approach.
+	var images []string
+	images = append(images, imageInspect.RepoTags...)
+	images = append(images, imageInspect.RepoDigests...)
+	images = append(images, image.Image)
+
+	for _, image := range images {
+		if _, err := ds.client.RemoveImage(image, dockertypes.ImageRemoveOptions{PruneChildren: true}); err != nil && !libdocker.IsImageNotFoundError(err) {
+			return nil, err
```
Comment (redbaron) [inline pkg/kubelet/dockershim/docker_image.go]: wouldn't it start returning image not found error after all tags are deleted?

**Comment by redbaron** [comment]
@corvus-ch , which problem does it solve? comment suggests that repo digests are preventing image from being deleted, but in my testing digests do not create any references, tags do and deleting all tags is enough to delete image

**Pull request merged**